kubernetes/updater/pkg/version/versionget.go

```diff
@@ -0,0 +1,40 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package version
+
+import "github.com/rogpeppe/go-internal/semver"
+
+type Getter interface {
+	GetVersion() (string, error)
+}
+
+func ValidVersionChange(current, next, _ string) bool {
+	// Cannot upgrade to a non-valid version
+	if !semver.IsValid(next) {
+		// log.Error
+		return false
+	}
+	switch semver.Compare(next, current) {
+	// No need to upgrade if version is the same
+	case 0:
+		return false
+		// Upgrade to newer version is always a "yes"
```

Comment:
It does not need to be a newer version, Rollbacks must also force the upgrade and the version is older.

Reply:
I initially implemented downgrade limitations, but it was removed following discussions in the RFD. I'll clear this part